gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-users/src/main/java/io/gravitee/am/gateway/handler/users/resources/consents/UserConsentEndpointHandler.java

```diff
@@ -52,11 +57,18 @@ public void get(RoutingContext context) {
      * Revoke specific consent for a user
      */
     public void revoke(RoutingContext context) {
+        final JWT accessToken = context.get(ConstantKeys.TOKEN_CONTEXT_KEY);
         final String userId = context.request().getParam("userId");
         final String consentId = context.request().getParam("consentId");
 
+        final var singleUserId = getUserIdFromSub(accessToken);
         getPrincipal(context)
-                .flatMapCompletable(principal -> userService.revokeConsent(userId, consentId, principal))
+                .flatMapCompletable(principal -> singleUserId.flatMapCompletable(id -> {
```

Comment:
What about inline `singleUserId` to getPrincipal stream, like this? 
```suggestion
                .flatMapCompletable(principal -> getUserIdFromSub(accessToken).flatMapCompletable(id -> {
```